spec/datadog/appsec/integration/waf_requests_telemetry_spec.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+require 'datadog/tracing/contrib/support/spec_helper'
+require 'datadog/appsec/spec_helper'
+require 'rack/test'
+
+require 'datadog/tracing'
+require 'datadog/appsec'
+
+RSpec.describe 'WAF requests telemetry' do
+  include Rack::Test::Methods
+
+  before do
+    Datadog.configure do |c|
+      c.tracing.enabled = true
+      c.tracing.instrument :rack
+      c.tracing.instrument :http
+
+      c.appsec.enabled = true
+      c.appsec.instrument :rack
+
+      c.appsec.waf_timeout = 10_000_000 # in us
+      c.appsec.ip_passlist = []
+      c.appsec.ip_denylist = []
+      c.appsec.user_id_denylist = []
+      c.appsec.ruleset = :recommended
+      c.appsec.api_security.enabled = false
+      c.appsec.api_security.sample_rate = 0.0
+
+      c.remote.enabled = false
+    end
+
+    allow_any_instance_of(Datadog::Tracing::Transport::HTTP::Client).to receive(:send_request)
+  end
+
+  after do
+    Datadog.configuration.reset!
+    Datadog.registry[:rack].reset_configuration!
+  end
+
+  let(:app) do
+    stack = Rack::Builder.new do
+      use Datadog::Tracing::Contrib::Rack::TraceMiddleware
+      use Datadog::AppSec::Contrib::Rack::RequestMiddleware
+
+      map '/waf' do
+        run ->(_env) { [200, {'Content-Type' => 'text/html'}, ['OK']] }
+      end
+    end
+
+    stack.to_app
+  end
+
+  describe 'appsec.waf.requests telemetry' do
+    subject(:response) { last_response }
```

Comment:
Is it used anywhere?

Reply:
fixed [e7a12fe](https://github.com/DataDog/dd-trace-rb/pull/4839/commits/e7a12fe04cab674ff7ed61d2a6c9d2fe57d81bb1)